packages/xo-server/src/api/vm.js

```diff
@@ -1134,7 +1134,7 @@ revert.params = {
 }
 
 revert.resolve = {
-  snapshot: ['snapshot', 'VM-snapshot', 'administrate'],
+  snapshot: ['snapshot', 'VM-snapshot', 'operate'],
```

Comment:
The issue says that the user needs to be *operator of the VM* and *viewer of the snapshot* but you're checking that he's *operator of the snapshot*. It's almost equivalent but we should check the exact permissions to be safe.

Reply:
fixed :heavy_check_mark: